### PR TITLE
Terraform example code optimized

### PR DESCRIPTION
**Pull request by marifse**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: examples/basic/main.tf [hunk 1]
```diff
@@ -18,7 +19,7 @@ module "slz_vpc" {
   source            = "../../"
   resource_group_id = module.resource_group.resource_group_id
   region            = var.region
-  name              = var.name
+  name              = "${var.prefix}-slz-vpc"
```
Comment (ocofaigh) [inline examples/basic/main.tf:22]: no this does't make sense. when you pass a value for prefix, it uses the prefix and name. So var.prefix should not be used here. Can you just set it to `name = vpc` ?

File: examples/default/main.tf [hunk 2]
```diff
@@ -39,7 +40,7 @@ module "slz_vpc" {
   source                                 = "../../"
   resource_group_id                      = module.resource_group.resource_group_id
   region                                 = var.region
-  name                                   = var.name
+  name                                   = "${var.prefix}-slz-vpc"
```
Comment (ocofaigh) [inline examples/default/main.tf:43]: same as other comment

File: examples/no-prefix/main.tf [hunk 3]
```diff
@@ -1,3 +1,10 @@
+##############################################################################
+#Local variables
+##############################################################################
+locals {
+  vpc_name = "noprefix-vpc"
```
Comment (ocofaigh) [inline examples/no-prefix/main.tf:5]: set it to `vpc_name = "vpc"` so it matches what was there before

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

**Comment by marifse** [comment]
@ocofaigh please check now

**Comment by ocofaigh** [comment]
```
TestRunNoPrefixExample 2024-04-03T16:19:55Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; ╷
│ Error: Value for undeclared variable
│ 
│ A variable named "name" was assigned on the command line, but the root
│ module does not declare a variable of that name. To use this value, add a
│ "variable" block to the configuration.
╵}
```

Test needs to be updated to no longer pass a value for name I guess

**Comment by marifse** [comment]
@ocofaigh fixed the variable issue, can you please rerun the pipeline. Thanks

**Comment by marifse** [comment]
@ocofaigh can you please try rerunning this too

**Comment by marifse** [comment]
@ocofaigh sorry to bother, but can you please re run the pipeline.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**